src/NUnitFramework/framework/Internal/Results/TestSuiteResult.cs

```diff
@@ -112,46 +194,66 @@ public override IList<ITestResult> Children
         /// <param name="result">The result to be added</param>
         public virtual void AddResult(ITestResult result)
         {
-            Children.Add(result);
+#if PARALLEL
+            var childrenAsConcurrentQueue = Children as ConcurrentQueue<ITestResult>;
+            if (childrenAsConcurrentQueue != null)
+                childrenAsConcurrentQueue.Enqueue(result);
+            else
+#endif
+            {
+                var childrenAsIList = Children as IList<ITestResult>;
+                if (childrenAsIList != null)
+                    childrenAsIList.Add(result);
+                else
+                    throw new NotSupportedException("cannot add results to Children");
```

Comment:
The problem is that Children is a virtual property, and can be overriden by derived classes.  TestCaseResult is one example.  Therefore, _children cannot be used, but Children has to be.  Since Children can be anything that implements IEnumerable, it has to be done this way,